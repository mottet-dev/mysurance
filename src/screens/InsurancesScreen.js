import React, { Component } from 'react';
import { View, Text } from 'react-native';
import { Header, Button } from 'react-native-elements';


import { backgroundColor, lightColor, strongColor } from '../constants';
import InsurancesList from '../components/InsurancesList/InsurancesList';

class InsurancesScreen extends Component {
    render() {
        const { 
            screenStyle, 
            headerOuterStyle,
            componentTitleStyle,
            buttonStyle,
            buttonTextStyle,
            buttonContainerStyle,
        } = styles;

        return (
            <View style={screenStyle}>
                <Header
                    outerContainerStyles={headerOuterStyle}
                    centerComponent={{ text: 'Mysurance', style: componentTitleStyle }}
                />
                
                <InsurancesList />

                <View style={buttonContainerStyle}>
                    <Button 
                        textStyle={buttonTextStyle}
                        buttonStyle={buttonStyle}
                        title="Add Insurance"
                        onPress={() => this.props.navigation.navigate('addinsurance')}
                    />
                </View>
            </View>
        );
    }
}

const styles = {
    screenStyle: {
        backgroundColor,
        flex: 1,
    },
    headerOuterStyle: {
        backgroundColor: strongColor,
    },
    componentTitleStyle: {
        color: lightColor,
        fontSize: 20
    },
    buttonStyle: {
        backgroundColor: strongColor,
        height: 60,
        borderRadius: 50,
    },
    buttonTextStyle: {
        fontSize: 18
    },
    buttonContainerStyle: {
        position: 'absolute',
        bottom: 20,
        right: 0,
        left: 0
    }
};

export default InsurancesScreen;