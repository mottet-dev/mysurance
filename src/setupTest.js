/*  
    Configure Enzyme to work with React 16 
    Is called before every test
*/

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

export default configure({ adapter: new Adapter() });