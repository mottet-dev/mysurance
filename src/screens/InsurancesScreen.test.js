import '../setupTest';
import React from 'react';
import { shallow } from 'enzyme';

import InsurancesScreen from './InsurancesScreen';

describe('<AddInsuranceScreen />', () => {
    const screen = shallow(<InsurancesScreen />);

    it('renders properly', () => {
        expect(screen).toMatchSnapshot();
    });
});