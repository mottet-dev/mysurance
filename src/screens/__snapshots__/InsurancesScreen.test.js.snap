// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddInsuranceScreen /> renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InsurancesScreen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Header
          centerComponent={
            Object {
              "style": Object {
                "color": "#FBF7F7",
                "fontSize": 20,
              },
              "text": "Mysurance",
            }
          }
          outerContainerStyles={
            Object {
              "backgroundColor": "#DC4712",
            }
          }
        />,
        <Connect(InsurancesList) />,
        <View
          style={
            Object {
              "bottom": 20,
              "left": 0,
              "position": "absolute",
              "right": 0,
            }
          }
        >
          <Button
            buttonStyle={
              Object {
                "backgroundColor": "#DC4712",
                "borderRadius": 50,
                "height": 60,
              }
            }
            onPress={[Function]}
            textStyle={
              Object {
                "fontSize": 18,
              }
            }
            title="Add Insurance"
          />
        </View>,
      ],
      "style": Object {
        "backgroundColor": "#F1E9E3",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "centerComponent": Object {
            "style": Object {
              "color": "#FBF7F7",
              "fontSize": 20,
            },
            "text": "Mysurance",
          },
          "outerContainerStyles": Object {
            "backgroundColor": "#DC4712",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Button
            buttonStyle={
              Object {
                "backgroundColor": "#DC4712",
                "borderRadius": 50,
                "height": 60,
              }
            }
            onPress={[Function]}
            textStyle={
              Object {
                "fontSize": 18,
              }
            }
            title="Add Insurance"
          />,
          "style": Object {
            "bottom": 20,
            "left": 0,
            "position": "absolute",
            "right": 0,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonStyle": Object {
              "backgroundColor": "#DC4712",
              "borderRadius": 50,
              "height": 60,
            },
            "onPress": [Function],
            "textStyle": Object {
              "fontSize": 18,
            },
            "title": "Add Insurance",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Header
            centerComponent={
              Object {
                "style": Object {
                  "color": "#FBF7F7",
                  "fontSize": 20,
                },
                "text": "Mysurance",
              }
            }
            outerContainerStyles={
              Object {
                "backgroundColor": "#DC4712",
              }
            }
          />,
          <Connect(InsurancesList) />,
          <View
            style={
              Object {
                "bottom": 20,
                "left": 0,
                "position": "absolute",
                "right": 0,
              }
            }
          >
            <Button
              buttonStyle={
                Object {
                  "backgroundColor": "#DC4712",
                  "borderRadius": 50,
                  "height": 60,
                }
              }
              onPress={[Function]}
              textStyle={
                Object {
                  "fontSize": 18,
                }
              }
              title="Add Insurance"
            />
          </View>,
        ],
        "style": Object {
          "backgroundColor": "#F1E9E3",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "centerComponent": Object {
              "style": Object {
                "color": "#FBF7F7",
                "fontSize": 20,
              },
              "text": "Mysurance",
            },
            "outerContainerStyles": Object {
              "backgroundColor": "#DC4712",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Button
              buttonStyle={
                Object {
                  "backgroundColor": "#DC4712",
                  "borderRadius": 50,
                  "height": 60,
                }
              }
              onPress={[Function]}
              textStyle={
                Object {
                  "fontSize": 18,
                }
              }
              title="Add Insurance"
            />,
            "style": Object {
              "bottom": 20,
              "left": 0,
              "position": "absolute",
              "right": 0,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "buttonStyle": Object {
                "backgroundColor": "#DC4712",
                "borderRadius": 50,
                "height": 60,
              },
              "onPress": [Function],
              "textStyle": Object {
                "fontSize": 18,
              },
              "title": "Add Insurance",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
